src/models/hooks/activityReport.js

```diff
@@ -185,7 +185,7 @@ const propagateSubmissionStatus = async (sequelize, instance, options) => {
         { goalTemplateId: goal.goalTemplateId },
         {
           where: { id: goal.id },
-          transaction: options.transaction,
+          // transaction: options.transaction,
```

Comment:
this will spread the goal updates from the main single transaction connection to a discreate connection attempt for each and every one.

Reply:
Yes, I know. I added these comments after our engineering meeting, and it sped up the occurrence of the issue. This PR isn't meant to be merged, it's merely a placeholder with tools to recreate the issue should someone want to